### ISSUE TITLE
Update to 1.5.3 on Fedora 20 broke Fedora20->RHEL6 connection

### ISSUE THREAD

**Issue by maxamillion** [ansible/ansible#6784]
##### Issue Type:

Bug Report
##### Ansible Version:

ansible 1.5.3 
##### Environment:

Running Ansible from Fedora 20, managing RHEL6 
##### Summary:

When using a Fedora 20 control machine to configure a RHEL6 host with ansible version 1.5.3, a playbook that previously worked no longer works claiming it is unable to connect to the remote host.

Playbook in question is hosted here: https://github.com/maxamillion/ansible-maxamillion.so

Following output:

```
(0)$ ansible-playbook -vvvvv site.yml -i inventory.txt -u root                                                       

PLAY [all] ******************************************************************** 

GATHERING FACTS *************************************************************** 
attempting connection to maxamillion.so via the accelerated port 5099
maxamillion.so: in recv_data(), waiting for the header
maxamillion.so: got the header, unpacking
maxamillion.so: data received so far (expecting 57): 0
maxamillion.so: received all of the data, returning
maxamillion.so: received the response
<maxamillion.so> REMOTE_MODULE setup
EXEC COMMAND mkdir -p $HOME/.ansible/tmp/ansible-tmp-1396302376.85-108604416462738 && echo $HOME/.ansible/tmp/ansible-tmp-1396302376.85-108604416462738
maxamillion.so: in recv_data(), waiting for the header
maxamillion.so: got the header, unpacking
maxamillion.so: data received so far (expecting 57): 0
maxamillion.so: received all of the data, returning
maxamillion.so: received the response
fatal: [maxamillion.so] => Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/ansible/runner/__init__.py", line 530, in _executor
    exec_rc = self._executor_internal(host, new_stdin)
  File "/usr/lib/python2.7/site-packages/ansible/runner/__init__.py", line 621, in _executor_internal
    return self._executor_internal_inner(host, self.module_name, self.module_args, inject, port, complex_args=complex_args)
  File "/usr/lib/python2.7/site-packages/ansible/runner/__init__.py", line 807, in _executor_internal_inner
    result = handler.run(conn, tmp, module_name, module_args, inject, complex_args)
  File "/usr/lib/python2.7/site-packages/ansible/runner/action_plugins/normal.py", line 57, in run
    return self.runner._execute_module(conn, tmp, module_name, module_args, inject=inject, complex_args=complex_args)
  File "/usr/lib/python2.7/site-packages/ansible/runner/__init__.py", line 409, in _execute_module
    tmp = self._make_tmp_path(conn)
  File "/usr/lib/python2.7/site-packages/ansible/runner/__init__.py", line 1015, in _make_tmp_path
    if result['rc'] != 0:
KeyError: 'rc'


TASK: [common | Install Packages] ********************************************* 
FATAL: no hosts matched or all hosts have already failed -- aborting


PLAY RECAP ******************************************************************** 
           to retry, use: --limit @/home/admiller/site.retry

maxamillion.so             : ok=0    changed=0    unreachable=1    failed=0

$ ssh maxamillion.so
[root@maxamillion ~]# mkdir -p $HOME/.ansible/tmp/ansible-tmp-1396302376.85-108604416462738 && echo $HOME/.ansible/tmp/ansible-tmp-1396302376.85-108604416462738
/root/.ansible/tmp/ansible-tmp-1396302376.85-108604416462738
[root@maxamillion ~]# exit
logout
Connection to maxamillion.so closed.

$ rpm -q ansible
ansible-1.5.3-2.fc20.noarch
```

If I run from the latest git checkout it works just fine:

```
$ ansible --version
ansible 1.6 (devel 33eec0dc55) last updated 2014/03/31 17:03:42 (GMT -500)
(0)$ ansible-playbook site.yml -i inventory.txt

PLAY [all] ********************************************************************

GATHERING FACTS ***************************************************************
ok: [maxamillion.so]

TASK: [common | Install Packages] *********************************************
ok: [maxamillion.so] => (item=python-keyczar,yum-utils,python-simplejson,libselinux-python)

TASK: [common | Enable SELinux - enforcing] ***********************************
ok: [maxamillion.so]

TASK: [common | Check for epel] ***********************************************
ok: [maxamillion.so]

TASK: [common | Copy RPM to machine] ******************************************
skipping: [maxamillion.so]

TASK: [common | Copy EPEL GPG Key] ********************************************
skipping: [maxamillion.so]

TASK: [common | Import EPEL GPG Key] ******************************************
skipping: [maxamillion.so]

TASK: [common | Install EPEL RPM] *********************************************
skipping: [maxamillion.so]

TASK: [common | Install Packages] *********************************************
ok: [maxamillion.so] => (item=fedora-packager,fedora-review,fedora-review-tests,mutt,nmap,sysstat,vim-enhanced,tmux,screen,git,rpm-build,mock,ruby,ruby-irb,ruby-devel,rubygems,man-pages,kernel-doc,rsync,wget,tito,rpmdevtools,rpmconf,htop,powertop,iotop,fedpkg,scl-utils,scl-utils-build,spec2scl,setools-console,strace,euca2ools,tig,lm_sensors,yum-plugin-security,autoconf,automake,libtool,intltool,gnupg,testdisk,tcpdump,httpd-tools,lsscsi,irssi,policycoreutils-python,setroubleshoot-server,ntp,ntpdate,bind-utils)

TASK: [common | Set TimeZone in sysconfig] ************************************
ok: [maxamillion.so]

TASK: [common | Set TimeZone in /etc/localtime] *******************************
ok: [maxamillion.so]

TASK: [common | Start and Enable ntpd] ****************************************
ok: [maxamillion.so]

TASK: [resume | Install httpd for resume hosting] *****************************
ok: [maxamillion.so] => (item=httpd)

TASK: [resume | Upload resume] ************************************************
ok: [maxamillion.so]

TASK: [resume | Start and enable httpd] ***************************************
ok: [maxamillion.so]

TASK: [users | Setup User account for admiller] *******************************
ok: [maxamillion.so]

TASK: [users | Ensure admiller ~/.ssh dir exists] *****************************
ok: [maxamillion.so]

TASK: [users | Ensure authorized_keys installed for admiller] *****************
ok: [maxamillion.so]

TASK: [users | Ensure file permissions for admiller authorized keys] **********
ok: [maxamillion.so]

TASK: [iptables | Copy iptables rules] ****************************************
ok: [maxamillion.so]

TASK: [iptables | Start and Enable iptables] **********************************
ok: [maxamillion.so]

PLAY RECAP ********************************************************************
maxamillion.so             : ok=17   changed=0    unreachable=0    failed=0
```
##### Steps To Reproduce:

Clone https://github.com/maxamillion/ansible-maxamillion.so, point the inventory at a RHEL6 machine from a Fedora 20 machine using ansible 1.5.3 
##### Expected Results:

I expected the playbook to work, just as it did from previous versions as well as from the latest git checkout.
##### Actual Results:

Actual results reported in output in the Summary section.

**Comment by mpdehaan** [comment]
Hmm, if this were generically true I think we'd get a lot more reports -- this may be an error message down an error path.

Have you tried this on the development branch?

Thanks!

**Comment by maxamillion** [comment]
@mpdehaan - I did try with the development branch, I even explicitly said in the issue report that it works with the development branch and included output showing the version and it working.

**Comment by jimi-c** [comment]
As discussed on IRC, since this is fixed in devel we're going to go ahead and close this. I'm having a bit of difficulty in pinning down the exact commit that fixed this, but regardless it will be included in the next major release. Thanks!

**Comment by rrix** [comment]
Hi there, I'm not sure what 'next release' is (major or minor) but in 1.5.4 I am having this issue, as well with Fedora 20=>CentOS6.5

**Comment by jimi-c** [comment]
@rrix if we end up doing another minor release (1.5.5) it will be included in that, otherwise it will be included in 1.6.
